### PR TITLE
Remove refcase in plotter

### PR DESCRIPTION
**Pull request by oysteoh**
**Approach**
Seems like retrieval of historical vectors either by refcase or from the realization itself is already supported in the code. In this PR i have removed everything related to the refcase in the plotter. In addition i have moved the retrieval from using libres_facade directly to utilize our already made endpoint for getting data from keys. What is missing as i see it is to generate historical vectors for all cases if they are NOT equal, but there is another issue for that - #3299 . 

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Review by DanSava — COMMENTED** [review]
Looks good :+1: , one question should `refcase_data` from `ert_shared/libres_facade.py` be also removed. Seems it is linked to  `history_data` but it might not really need to be. Also used in two other tests:
`test_case_refcase_data`  & test_case_refcase_data_missing_key both in `tests/ert_tests/shared/test_libres_facade.py`

**Comment by oysteoh** [comment]
> Looks good +1 , one question should `refcase_data` from `ert_shared/libres_facade.py` be also removed. Seems it is linked to `history_data` but it might not really need to be. Also used in two other tests: `test_case_refcase_data` & test_case_refcase_data_missing_key both in `tests/ert_tests/shared/test_libres_facade.py`

I will have a look, might be :+1:

**Comment by oysteoh** [comment]
> Looks good +1 , one question should `refcase_data` from `ert_shared/libres_facade.py` be also removed. Seems it is linked to `history_data` but it might not really need to be. Also used in two other tests: `test_case_refcase_data` & test_case_refcase_data_missing_key both in `tests/ert_tests/shared/test_libres_facade.py`

`has_refcase` i can remove since it is not used, but `refcase_data` i'll leave as-is. The point here is that users using the refcase should have the history vector from the refcase plotted. If a refcase or a history-vector from the refcase does not exist - we should get the history-vector from the summary-data.

Did a minor refactoring changing the variable-name `key` -> `history_key` when assigned to make it more clear - i did not like the reuse of the argument to `history_data` function.

File: ert_gui/tools/plot/plot_api.py [hunk 1]
```diff
@@ -209,12 +205,13 @@ def _add_index_range(self, data):
...
-        return self._facade.refcase_data(key)
-
     def history_data(self, key, case=None):
         """Returns a pandas DataFrame with the data points for the history for a given data key, if any.
         The row index is the index/date and the column index is the key."""
-        return self._facade.history_data(key, case)
+
+        if ":" in key:
+            head, tail = key.split(":", 2)
+            history_key = "{}H:{}".format(head, tail)
```
Comment (berland) [inline ert_gui/tools/plot/plot_api.py:214]: Pylint will complain here if you don't use `f-strings`.

Comment (oysteoh) [reply ert_gui/tools/plot/plot_api.py:214]: might be - just moved old code from one function to another here - haven't made this myself. I will have a look at it :)

**Comment by frode-aarstad** [comment]
Did a small comparison between stable and this branch. 
image

**Review by frode-aarstad — APPROVED** [review]
LGTM

**Pull request merged**